document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -985,7 +985,11 @@ protected Document prepareDocument() throws SQLException, IOException {
       Map<String, Object> jsonNode = new HashMap();
       for (int i = 1; i <= columnCount; i++) {
         String columnName = resultSetMetaData.getColumnName(i);
-        String columnValue = resultSet.getString(i);
+        int columnType = resultSetMetaData.getColumnType(i);
+        String columnValue =
+            columnType == 2003
```

Comment:
2003 means java.sql.array type

Reply:
I meant defining an integer constant `ARRAY_COLUMN_TYPE = 2003` and using it here. 

Can be done in a different PR.